specs/WebNotification.md

```diff
@@ -0,0 +1,610 @@
+}
+
+/// A collection of unsigned long integers.
+[uuid(974A91F8-A309-4376-BC70-7537D686533B), object, pointer_default(unique)]
+interface ICoreWebView2UnsignedLongCollection : IUnknown {
+  /// The number of unsigned long integers contained in the
+  /// ICoreWebView2UnsignedLongCollection.
+  [propget] HRESULT Count([out, retval] UINT* count);
+
+  /// Gets the unsigned long integer at the given index.
+  HRESULT GetValueAtIndex([in] UINT index, [out, retval] UINT64* value);
+}
+
+/// This is the ICoreWebView2Notification that represents a [HTML Notification
+/// object](https://developer.mozilla.org/docs/Web/API/Notification).
+[uuid(E3F43572-2930-42EB-BD90-CAC16DE6D942), object, pointer_default(unique)]
+interface ICoreWebView2Notification : IUnknown {
+  /// Add an event handler for the `Closed` event.
+  /// This event is raised when the notification is closed by the web code.
+  HRESULT add_Closed(
+      [in] ICoreWebView2NotificationClosedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_Closed`.
+  HRESULT remove_Closed(
+      [in] EventRegistrationToken token);
+
+  /// The host may run this to report the notification has been displayed.
+  /// The NotificationReceived event is considered handled regardless of the
+  /// Handled property of the NotifificationReceivedEventArgs if the host has
+  /// run Show().
+  HRESULT Show();
+
+  /// The host may run this to report the notification has been clicked. Use
+  /// `ClickWithAction` to specify an action to activate a persistent
+  /// notification. This will no-op if `Show` is not run or `Close` has been
+  /// run.
+  HRESULT Click();
+
+  /// The host may run this to report the persistent notification has been
+  /// activated with a given action. The action index corresponds to the index
+  /// in NotificationActionCollection. This returns `E_INVALIDARG` if an invalid
+  /// action index is provided. Use `Click` to activate an non-persistent
+  /// notification. This will no-op if `Show` is not run or `Close` has been
+  /// run.
+  HRESULT ClickWithAction([in] UINT actionIndex);
+
+  /// The host may run this to report the notification was dismissed.
+  /// This will no-op if `Show` is not run or `Click` has been run.
+  HRESULT Close();
+
+  /// The body string of the notification as specified in the constructor's
+  /// options parameter.
+  /// The default value is an empty string.
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`.  See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  [propget] HRESULT Body ([out, retval] LPWSTR* body);
+
+  /// Returns an IDataObject that represents a structured clone of the
+  /// notification's data.
+  /// Returns `null` if the optional Notification property does not exist.
+  [propget] HRESULT Data ([out, retval] IDataObject** data);
```

Comment:
How do we expect this to be used? In the MDN docs for notification. Data the example they have is of a simple string. Some other options to consider:
DataAsString - the data value's string value if data is a string
DataAsJson - the data value converted to JSON
DataAsDataObject

We could do all three of the above. But I don't know what this is really used for so I don't really know. Is this an uncommon parameter? If not, we should have some sample code for it.

Please also update the ref docs to describe what this is used for.